core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -116,6 +117,136 @@ final class Stream[+F[_], +O] private (private val free: FreeC[Algebra[Nothing,
+    *       , worker.evalMap { o => IO.println(s"3:$o") }
+    *     )
+    *   }.parJoinUnbounded.compile.drain.unsafeRunSync
+    * }}}
+    *
+    * Note that in the example above the workers are not guaranteed to see all elements emitted. This is
+    * due different subscription time of each worker and speed of emitting the elements by `this`.
+    *
+    * If this is not desired, consider using `broadcastTo` or `broadcastThrough` alternatives.
+    *
+    * When `this` terminates, the resulting streams (workers) are terminated once all elements so far pulled
+    * from `this` are processed by all workers.
+    *
+    *
+    * @tparam F2
+    * @return
+    */
+  def broadcast[F2[x] >: F[x]: Concurrent]: Stream[F2, Stream[F2, O]] =
+    Broadcast[F2, O](this)
+
+  /**
+    * Like `broadcast` but instead of providing stream as source for worker, it runs each worker through
+    * supplied sink.
+    *
+    * Each supplied sink is run concurrently with other. This means that amount of sinks determines parallelism.
+    * Each sink may have different implementation, if required, for example one sink may process elements, the other
+    * may send elements for processing to another machine.
+    *
+    * Also this guarantees, that each sink will view all `O` pulled from `this`.
+    *
+    * Note that resulting stream will not emit single value (Unit). If you need to emit unit values from your sinks, consider
+    * using `broadcastThrough`
+    *
+    * @param sinks    Sinks that will concurrently process the work.
+    * @tparam F2
+    * @return
+    */
+  def broadcastTo[F2[x] >: F[x]: Concurrent](sinks: Sink[F2, O]*): Stream[F2, Unit] =
+    broadcastThrough[F2, Nothing](sinks.map { _.andThen { _.drain } }: _*)
+
+  /**
+    * Variant of `broadcastTo` that takes number of concurrency required and single sink
+    * @param maxConcurrent   Maximum number of sinks to run concurrently
+    * @param sink          Sink to use to process elements
+    * @return
+    */
+  def broadcastTo[F2[x] >: F[x]: Concurrent](maxConcurrent: Int)(
+      sink: Sink[F2, O]): Stream[F2, Unit] =
+    this.broadcastTo[F2]((0 until maxConcurrent).map(_ => sink): _*)
+
+  def broadcastThrough[F2[x] >: F[x]: Concurrent, O2](pipes: Pipe[F2, O, O2]*): Stream[F2, O2] =
+    Stream
+      .eval {
+        Deferred[F2, Unit].flatMap { start =>
+          Semaphore[F2](0).map { started =>
+            // relies on fact that next element won't be pulled from source until all subscribers
+            // will commit. As such we just take first source, that asks for the first element
+            // and then start all remaining  sinks
+            // when all registers their start, we just terminate the first and continue with all remianing.
+
+            if (pipes.isEmpty) Stream.empty
+            else {
+              this.broadcast[F2].flatMap { source =>
+                def subscribe(pipe: Pipe[F2, O, O2]) =
+                  source.pull.uncons1
```

Comment:
Why operate on individual elements here (uncons1) instead of chunks?

Reply:
I am not sure on this one. I had it implemented with chunks, essentially saying that `if you need to operate on single elements, rechunk to single element chunks`.  Then, I sort of for consistency did same what we have on queue, so operate on  element and leave chunking to user. 

I have no strong preference here, whether chunking operation has to be explicit or implicit.